### PR TITLE
replace pl_lowercase and pl_uppercase with std::towlower and std::towupper

### PR DESCRIPTION
**Pull request by bradallred**
@MarcelHB could you check this out with German? you might need to change your locale to match the TLK. We can do that programmatically later if needed.

### CODE REVIEW

**Comment by lynxlynxlynx** [comment]
default_ini.txt can go as well.

**Comment by MarcelHB** [comment]
Any idea what to look at in particular? It's just that I see working things for that I don't know if they are a sign of success, and stuff that looks bad but it's not related to this here.

**Comment by bradallred** [comment]
looking for behavioral differences between this branch and master pertaining to lowercasing and uppercasing. It might be easier to check by editing the code to do something like force add `IE_GUI_BUTTON_CAPS` to all buttons to test uppercase, then change it to `IE_GUI_BUTTON_LOWERCASE` to test lowerccase.

**Review by MarcelHB — APPROVED** [review]
I forced button strings to go into either direction and did not discover any issues.

**Comment by bradallred** [comment]
great, thanks! Did you manage to test strings containing chars such as ü?

**Comment by MarcelHB** [comment]
Yeah, looks good.

File: gemrb/plugins/ITMImporter/ITMImporter.cpp [hunk 1]
```diff
@@ -109,7 +109,7 @@ bool ITMImporter::Import(DataStream* str)
 static void AddZZFeatures(Item *s)
 {
 	// the targeting code (3rd char) is: digit = align(ment), letter = race
-	char targetIDS = toupper(s->Name.CString()[2]);
+	char targetIDS = towupper(s->Name.CString()[2]);
```
Comment (lynxlynxlynx) [inline gemrb/plugins/ITMImporter/ITMImporter.cpp:112]: how does this even compile, doesn't towupper return a wchar_t? :s

**Comment by bradallred** [comment]
Thanks @MarcelHB, you've been a tremendous help.

I've chatted `zzmap` to use `wchar_t` as well

**Comment by lynxlynxlynx** [comment]
default_ini.txt still needs to be updated.

**Comment by bradallred** [comment]
updated or deleted? I dont see anything in there particularly relevant anymore

**Comment by lynxlynxlynx** [comment]
Ideally we'd document the remaining format (sections and keys), since they're not mentioned anywhere.

**Pull request merged**